### PR TITLE
fix: before revoking guest users, wait for removing guest link [AR-3310]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3310" title="AR-3310" target="_blank">BugAR-3310</a>  Removing guests did not remove all guests (in UI only?)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Because removing guest link and revoking guests access are triggered asynchronously which may lead to failed request when removing guest link will be triggered after revoking guests access

### Causes (Optional)

I don't know the context of this function but I assume that there was issue that we didn't removed guest link before revoking guests access. CC @tmspzz 

### Solutions

Wait for removing guest link response and then revoke guests access

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6301757)(6301757)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6301757)(6301757)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt [hunk 1]
```diff
@@ -224,19 +224,19 @@ class EditGuestAccessViewModel @Inject constructor(
     }
 
     fun onRevokeDialogConfirm() {
-        removeGuestLink()
+        viewModelScope.launch {
+            removeGuestLink()
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt]: should we move this the kalium usecase 

1. revoke guests access
2. on success revoke guest link

**Review by tmspzz — APPROVED** [review]

**Review by ohassine — COMMENTED** [review]
But isn't the same ? you moved starting a new coroutine from `removeGuestLink()` to `onRevokeDialogConfirm()`

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6335656)(6335656)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6335656)(6335656)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt]: Ok, but revoke should be before revoking guest access

**Comment by Garzas** [comment]
> But isn't the same ? you moved starting a new coroutine from `removeGuestLink()` to `onRevokeDialogConfirm()`

No, because now it is suspend and needs to wait until it completes

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6347591)(6347591)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6347591)(6347591)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6370130)(6370130)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6370130)(6370130)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/7/) **failed**.

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6405258)(6405258)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6405258)(6405258)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/12/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6405587)(6405587)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6405587)(6405587)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1694/13/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6405801)(6405801)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6405801)(6405801)-staging-compat.apk)

**Pull request merged**